In [2]:
#Import libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy

In [3]:
#Data import
df = pd.read_csv("data.csv")
df = df.drop(['Rank'] , axis = 1) #Removing 1 unnecessary column

In [4]:
df.head(10)

Name Platform    Year         Genre Publisher  \
0                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5                     Tetris       GB  1989.0        Puzzle  Nintendo   
6      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7                   Wii Play      Wii  2006.0          Misc  Nintendo   
8  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Name          16598 non-null  object 
 1   Platform      16598 non-null  object 
 2   Year          16327 non-null  float64
 3   Genre         16598 non-null  object 
 4   Publisher     16540 non-null  object 
 5   NA_Sales      16598 non-null  float64
 6   EU_Sales      16598 non-null  float64
 7   JP_Sales      16598 non-null  float64
 8   Other_Sales   16598 non-null  float64
 9   Global_Sales  16598 non-null  float64
dtypes: float64(6), object(4)
memory usage: 1.3+ MB


In [6]:
df.describe()

Year      NA_Sales      EU_Sales      JP_Sales   Other_Sales  \
count  16327.000000  16598.000000  16598.000000  16598.000000  16598.000000   
mean    2006.406443      0.264667      0.146652      0.077782      0.048063   
std        5.828981      0.816683      0.505351      0.309291      0.188588   
min     1980.000000      0.000000      0.000000      0.000000      0.000000   
25%     2003.000000      0.000000      0.000000      0.000000      0.000000   
50%     2007.000000      0.080000      0.020000      0.000000      0.010000   
75%     2010.000000      0.240000      0.110000      0.040000      0.040000   
max     2020.000000     41.490000     29.020000     10.220000     10.570000   

       Global_Sales  
count  16598.000000  
mean       0.537441  
std        1.555028  
min        0.010000  
25%        0.060000  
50%        0.170000  
75%        0.470000  
max       82.740000

# Column analysis based on Descriptive Statistics:
- `Year`: The Year column contains 16327 records, while the other columns contain 16598, which may indicate that in 271 cases there is no information about the year of manufacture, need to check the Column for missing values, and fill in/delete
  
- `NA_Sales`: Check the column for abnormal outliers, as well as check the distribution of sales and determine which games made the main cash

  
- `EU_Sales`: Need to compare the column with NA_Sales what genres or platforms are more popular in Europe. Are there any zero values that should be processed

  
- `JP_Sales`: Need to see which platforms are dominant and whether there are any strong differences between regions

  
- `Other_Sales`: It is necessary to determine which regions are included in this column, and also to study the dependence on global sales

  
- `Global_Sales`: Check the correlation with regional sales. Find out which games sell well everywhere, and which are popular only in certain regions

In [7]:
df.dtypes

Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

# Analysis of Column Type
- `Column Name` : Name of the Game(-s) , "object" type so correct. Need to check for duplicates
  
- `Platform` : Name of Platform where the games were released, "object" type , correct as well.

   
- `Year` : Year when games were published , type is "float64" which incorrect, year can't be float values, needed to be convert into integer values.

  
- `Genre`: Genre of Games(Platform,Sports,Racing,etc.) , "object" type correct.

  
- `Publisher`: Publisher of the game , "object" type correct.

  
- `NA_Sales`: Sales in NA regiom , "float64" correct

  
- `EU_Sales`: Sales in EU , "float64" dtype is valid

  
- `JP_Sales`: Sales in Japan , "float64" valid

  
- `Other_Sales`: Sales in other regions , "float64" is valid as for other regions

  
- `Global_Sales`: Worldwide sales , "fload64" is correct

In [8]:
df.isnull().sum()

Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [9]:
df['Name'].nunique()

11493

In [13]:
missing_values = df.loc[df['Year'].isna(), ['Name' , 'Platform' , 'Year']]
print(missing_values)

                              Name Platform  Year
179                Madden NFL 2004      PS2   NaN
377               FIFA Soccer 2004      PS2   NaN
431     LEGO Batman: The Videogame      Wii   NaN
470     wwe Smackdown vs. Raw 2006      PS2   NaN
607                 Space Invaders     2600   NaN
...                            ...      ...   ...
16307                Freaky Flyers       GC   NaN
16327                    Inversion       PC   NaN
16366  Hakuouki: Shinsengumi Kitan      PS3   NaN
16427                 Virtua Quest       GC   NaN
16493                   The Smurfs      3DS   NaN

[271 rows x 3 columns]


In [41]:
duplicates = df.loc[df.duplicated(subset = ['Name' , 'Platform'] , keep=False), ['Name', 'Platform' , 'Year']]
print(duplicates)
print(len(duplicates))
print(df.loc[df['Name'] == 'Need for Speed: Most Wanted', ['Name','Platform','Year']])
print(df.loc[df['Name'] == 'Need for Speed: Most Wanted'].index)

                              Name Platform    Year
603                  Madden NFL 13      PS3  2012.0
1715            Sonic the Hedgehog      PS3  2006.0
4145            Sonic the Hedgehog      PS3     NaN
5900   Need for Speed: Most Wanted       PC  2005.0
11676  Need for Speed: Most Wanted       PC  2012.0
14997  Wii de Asobu: Metroid Prime      Wii     NaN
14999  Wii de Asobu: Metroid Prime      Wii     NaN
16127                Madden NFL 13      PS3  2012.0
8
                              Name Platform    Year
252    Need for Speed: Most Wanted      PS2  2005.0
498    Need for Speed: Most Wanted      PS3  2012.0
1530   Need for Speed: Most Wanted     X360  2005.0
1742   Need for Speed: Most Wanted      PSV  2012.0
2005   Need for Speed: Most Wanted       XB  2005.0
3585   Need for Speed: Most Wanted       GC  2005.0
5900   Need for Speed: Most Wanted       PC  2005.0
6149   Need for Speed: Most Wanted     WiiU  2013.0
6278   Need for Speed: Most Wanted       DS  2005.0
6492   Nee